### PR TITLE
Fix issue with products GET request being cached despite category positions change

### LINKED ISSUE #30467: Products positions are not updated in GraphQL query result
#### Precondition
Use GET to submit GraphQL query in order for query result to be cached. see [GraphQL Caching](https://devdocs.magento.com/guides/v2.4/graphql/caching.html)

#### Steps to reproduce
1. Create at least 2 products
1. Create category and assign products created in step.1 and save category
1. Execute the query below
1. Check that products are ordered correctly
1. Change products positions in the category and save changes
1. Execute the query below

#### Query
```java
query getProducts { 
  products( 
    filter: {
      category_id: 
        { eq: "<category ID>" } 
    } 
    sort:
    	{position:ASC} 
  ) 
  { 
  total_count 
    items { 
      name 
      sku 
    } 
    page_info { 
      page_size 
      current_page 
    } 
  } 
}
```
#### (x) Actual result

Products positions are not updated.
#### (/) Expected result

Products positions are updated.

#### (i) Notice
This issue is reproducible only if GraphQL query result is cached. the cache is not invalidated when products positions changed within a category. However this works with regular category page (PLP).

### PR DESCRIPTION
**Pull request by Zifius**
### Description (*)

Fixes issue with products GET request being cached despite category positions change

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30467

### Manual testing scenarios (*)

Described in the original issue

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rogyar** [comment]
@Zifius please, feel free to ping me if you need any help with that.

**Comment by cpartica** [comment]
@Zifius  your fix looks good, you should open the PR and complete a test.
I think there's some integration tests done on tags as inspiration

File: app/code/Magento/CatalogGraphQl/Model/Resolver/Products.php [hunk 1]
```diff
@@ -79,6 +79,11 @@ public function resolve(
             'layer_type' => isset($args['search']) ? Resolver::CATALOG_LAYER_SEARCH : Resolver::CATALOG_LAYER_CATEGORY,
         ];
 
+        if (isset($args['filter'])) {
```
Comment (cpartica) [inline app/code/Magento/CatalogGraphQl/Model/Resolver/Products.php]: you only need some tests, fix looks good

**Review by cpartica — CHANGES_REQUESTED** [review]

**Comment by Zifius** [comment]
Thanks @cpartica, unfortunately community slack doesn't keep the history of our 1to1 chat, so the question is would you prefer changing \Magento\GraphQl\Catalog\ProductSearchTest::testSortByPosition() (https://github.com/magento/magento2/blob/2.4-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/ProductSearchTest.php#L1114) which seems to be testing exactly this functionality but currently passing (I suppose the test is incomplete or incorrect, but need to debug it to see what's really going on there) or creating a new test?

**Comment by rogyar** [comment]
Just from my point of view, I would suggest adjusting an existing test case instead of introducing a new one with a very similar purpose. 

Not sure my assumption makes sense. But the existing test might work and the reason for that is utilizing disk/Redis cache for FPC instead of Varnish. But I might be wrong.

**Comment by cpartica** [comment]
> Just from my point of view, I would suggest adjusting an existing test case instead of introducing a new one with a very similar purpose.
> 
> Not sure my assumption makes sense. But the existing test might work and the reason for that is utilizing disk/Redis cache for FPC instead of Varnish. But I might be wrong.

Yes we can't test Varnish yet, but we sure can do disk builtin cache with the help of integration tests

**Comment by Zifius** [comment]
Thank you @rogyar and @cpartica both for the replies. I've now created the test

**Comment by Zifius** [comment]
@magento run WebAPI Tests

**Comment by rogyar** [comment]
@magento run WebAPI Tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by Zifius** [comment]
@magento run Integration Tests

**Comment by Zifius** [comment]
@magento run all tests

**Review by rogyar — APPROVED** [review]
Approved. The failing functional tests are not related to the current changeset.

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8504](https://jira.corp.magento.com/browse/ENGCOM-8504) has been created to process this Pull Request

**Review by cpartica — APPROVED** [review]

**Pull request merged**